engine/src/main/java/org/entando/entando/web/userprofile/ProfileTypeController.java

```diff
@@ -0,0 +1,150 @@
+import com.agiletec.aps.system.exception.ApsSystemException;
+import com.agiletec.aps.system.services.role.Permission;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import javax.validation.Valid;
+import org.entando.entando.aps.system.exception.RestRourceNotFoundException;
+import org.entando.entando.aps.system.services.entity.model.EntityTypeShortDto;
+import org.entando.entando.aps.system.services.userprofile.IUserProfileTypeService;
+import org.entando.entando.aps.system.services.userprofile.model.UserProfileTypeDto;
+import org.entando.entando.web.common.annotation.RestAccessControl;
+import org.entando.entando.web.common.exceptions.ValidationConflictException;
+import org.entando.entando.web.common.exceptions.ValidationGenericException;
+import org.entando.entando.web.common.model.PagedMetadata;
+import org.entando.entando.web.common.model.RestListRequest;
+import org.entando.entando.web.common.model.RestResponse;
+import org.entando.entando.web.userprofile.model.ProfileTypeDtoRequest;
+import org.entando.entando.web.userprofile.validator.ProfileTypeValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author E.Santoboni
+ */
+@RestController
+@RequestMapping(value = "/profileTypes")
+public class ProfileTypeController {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
+    private IUserProfileTypeService userProfileTypeService;
+
+    @Autowired
+    private ProfileTypeValidator profileTypeValidator;
+
+    protected IUserProfileTypeService getUserProfileTypeService() {
+        return userProfileTypeService;
+    }
+
+    public void setUserProfileTypeService(IUserProfileTypeService userProfileTypeService) {
+        this.userProfileTypeService = userProfileTypeService;
+    }
+
+    protected ProfileTypeValidator getProfileTypeValidator() {
+        return profileTypeValidator;
+    }
+
+    public void setProfileTypeValidator(ProfileTypeValidator profileTypeValidator) {
+        this.profileTypeValidator = profileTypeValidator;
+    }
+
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getUserProfileTypes(RestListRequest requestList) throws JsonProcessingException {
```

Comment:
Why is this <?> and not < UserProfileTypeDto>

Reply:
The <?> is org.entando.entando.web.common.model.RestResponse, the object that contains the main payload, errors and metadata. Fixed!